source/shared/core_conn.cpp

```diff
@@ -150,34 +150,125 @@ sqlsrv_conn* core_sqlsrv_connect( _In_ sqlsrv_context& henv_cp, _In_ sqlsrv_cont
     
     build_connection_string_and_set_conn_attr( conn, server, uid, pwd, options_ht, valid_conn_opts, driver, conn_str TSRMLS_CC );
     
-    bool is_missing_driver = false;
+    if( conn->ce_option.enabled ) {
+        // when column encryption is enabled, must use ODBC driver 17
+        if( conn->driver_version != ODBC_DRIVER_UNKNOWN ) {
+            CHECK_CUSTOM_ERROR( conn->driver_version != ODBC_DRIVER_17, conn, SQLSRV_ERROR_CE_DRIVER_REQUIRED, get_processor_arch() ) {
+                throw core::CoreException();
+            }
+#ifndef _WIN32
```

Comment:
eww this ifndef is inside a nested if and there are multiple conditions after this before the else then it gets back to this condition and indent level. Looks very messy hard to read and maintain.
The condition it's checking before this simply throws an except and exits anyway why even bother with the else and all the indentation?
ie if (ce enabled && (driver unknown || driver doesn't exist) die

Reply:
the thing is the way to check if driver exists or not is very different in Windows and in Linux/mac...
it was you who insisted that I moved these few lines outside of ifndef... I already said it was harder to read

Reply:
the original comment I suggested was to refactor common c&p code outside the ifdef, not nest it to make it harder to read.  If there's any misunderstanding, don't make any changes that doesn't make any sense to you.

The block in green above is common in all platforms and results in exception being thrown.
Why not just:

if (ce enabled && !driver 17)
  throw

#ifndef WIN 
//do this
#else 
//do that